From d272dfdd4efb2072946228ea57818e76173e3d7c Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Sun, 29 May 2016 15:51:42 +0300
Subject: [PATCH] KLIPS: fix sk pointer usage for key_pid

up till LKv4.1, key_pid was saved into sk_proinfo, this field
was removed and KLIPS was changed to save the key_pid into
the sk pointer which crashed the kernel. This patch fixes this
issue by saving the key_pid into sk->sk_user_data.

Signed-off-by: Ofer Heifetz <oferh@marvell.com>
---
 linux/include/libreswan/pfkey.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux/include/libreswan/pfkey.h b/linux/include/libreswan/pfkey.h
index 3f70e15..29719f2 100644
--- a/linux/include/libreswan/pfkey.h
+++ b/linux/include/libreswan/pfkey.h
@@ -141,7 +141,7 @@ struct key_opt {
 };
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0)
-# define key_pid(sk) ((struct key_opt*)&(sk))->key_pid
+# define key_pid(sk) ((struct key_opt*)&((sk)->sk_user_data))->key_pid
 #else
 # define key_pid(sk) ((struct key_opt*)&((sk)->sk_protinfo))->key_pid
 #endif
-- 
2.8.2

